### PR TITLE
frontend: Added error Page

### PR DESCRIPTION
**Pull request by PrathamX595**
Added Error Page

Fixes: #21 

## Description
added an error page which is rendered when an unknown route is accessed, it contains a link to go  back to home page of the game.

## How to Test

1. use the command `npm run dev` in the console and open the application.
2. In the search box add `/k` (or any other route which is not recognized by the router) after the existing text.

## Checklist
- [x] I have tested these changes locally.
- [x] I have reviewed the code and ensured it follows the project's coding guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have assigned reviewers to this pull request.

## Screenshots (if applicable)
70e10a13-76c8-4561-8532-b4486e8272bd

### CODE REVIEW

**Comment by kuv2707** [comment]
@PrathamX595 Please remove the merge commit and use rebase to sync your branch with upstream/master.

**Comment by PrathamX595** [comment]
@kuv2707 I am having trouble removing the merge commit can someone assist me ?

**Comment by PrathamX595** [comment]
@kuv2707 I have implemented the recommended let me know if anything else needs to be done.

File: backend/src/types.d.ts [hunk 1]
```diff
@@ -27,6 +27,10 @@ type EventResult = {
     message: string;
 };
 
+declare global {
```
Comment (kuv2707) [inline backend/src/types.d.ts:30]: Please remove all changes to this file.

**Pull request closed without merging**